Don't define the macro if there's no backend support, it breaks the
proper fallback in uwac_create_anonymous_file().

Index: uwac/libuwac/uwac-os.c
--- uwac/libuwac/uwac-os.c.orig
+++ uwac/libuwac/uwac-os.c
@@ -43,7 +43,7 @@
 #endif
 
 /* uClibc and uClibc-ng don't provide O_TMPFILE */
-#if !defined(O_TMPFILE) && !defined(__FreeBSD__)
+#if !defined(O_TMPFILE) && !defined(__FreeBSD__) && !defined(__OpenBSD__)
 #define O_TMPFILE (020000000 | O_DIRECTORY)
 #endif
 
